identity/aziot-identity-client-async/src/lib.rs

```diff
@@ -4,37 +4,212 @@
 #![warn(clippy::all, clippy::pedantic)]
 #![allow(clippy::must_use_candidate, clippy::missing_errors_doc)]
 
+use aziot_identity_common::{Identity, ID_TYPE_AZIOT, ID_TYPE_LOCAL};
+
+// All exports of aziot_identity_common_http are used in this file.
+#[allow(clippy::wildcard_imports)]
+use aziot_identity_common_http::*;
+
+macro_rules! make_uri {
+    ($path:literal, $api_version:expr) => {
+        &format!(
+            "http://identityd.sock{}?api-version={}",
+            $path, $api_version
+        )
+    };
+    ($path:literal, $api_version:expr, $type:expr) => {
+        &format!(
+            "http://identityd.sock{}?api-version={}&type={}",
+            $path, $api_version, $type
+        )
+    };
+    ($path:literal, $api_version:expr, $type:expr, $name:ident) => {
+        &format!(
```

Comment:
You need to URL-encode the components, at least the `$name` since it's arbitrary.

Reply:
Good catch. Fixed.